File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int n=s.size();
        string temp="";
        string ans="";
        int i=0;
        while(i<n){
            temp="";
            while(i<n && s[i]==' ') i++;
            while(i<n && s[i]!=' '){
                temp+=s[i];
                i++;
            }
            if(!temp.empty()){
                if(ans.empty()) ans+=temp;
                else ans=temp+' '+ans;
            }
        }
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        n = len(s)
        ans = ""
        i = 0
        while i<n:
            word = ""
            while i<n and s[i] == ' ':
                i+=1
            while i<n and s[i]!=' ':
                word+=s[i]
                i+=1
            if len(word)>0:
                ans = word + " "+ans
        return ans.strip()
            
            

        